tests/extensions/webgpu_extension.h

```diff
@@ -0,0 +1,464 @@
+/**
+ * Copyright 2025 WebGPU-Native developers
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+/** @file */
+
+/**
+ * \mainpage
+ *
+ * This is a bare minimum extensions file to verify all extension types.
+ */
+
+#ifndef WEBGPU_EXTENSION_H_
+#define WEBGPU_EXTENSION_H_
+
+#include "webgpu.h"
+
+#if !defined(_wgpu_EXTEND_ENUM)
+#ifdef __cplusplus
+#define _wgpu_EXTEND_ENUM(E, N, V) static const E N = E(V)
+#else
+#define _wgpu_EXTEND_ENUM(E, N, V) static const E N = (E)(V)
+#endif
+#endif // !defined(_wgpu_EXTEND_ENUM)
+
+/**
+ * \defgroup Constants Constants
+ * \brief Constants.
+ *
+ * @{
+ */
+/**
+ * New constant should have the namespace prefix in the name by default.
+ */
+#define WGPU_PREFIX_NEW_CONSTANT1 (UINT32_MAX)
+/**
+ * New constant can be declared in the 'webgpu' namespace explicitly.
+ */
+#define WGPU_NEW_CONSTANT2 (UINT64_MAX)
+
+/** @} */
+
+/**
+ * \defgroup UtilityTypes Utility Types
+ *
+ * @{
+ */
+
+/**
+ * New typedef should have the namespace prefix in the name by default.
+ */
+typedef uint32_t WGPUPrefixNewTypedef1;
+/**
+ * New typedef can be declared in the 'webgpu' namespace explicitly.
+ */
+typedef uint64_t WGPUNewTypedef2;
+
+/** @} */
+
+/**
+ * \defgroup Objects Objects
+ * \brief Opaque, non-dispatchable handles to WebGPU objects.
+ *
+ * @{
+ */
+/**
+ * New object should have the namespace prefix in the name by default.
+ */
+typedef struct WGPUPrefixNewObject1Impl* WGPUPrefixNewObject1 WGPU_OBJECT_ATTRIBUTE;
+/**
+ * New object can be declared in the 'webgpu' namespace explicitly.
+ */
+typedef struct WGPUNewObject2Impl* WGPUNewObject2 WGPU_OBJECT_ATTRIBUTE;
+
+/** @} */
+
+// Structure forward declarations
+struct WGPUPrefixNewStruct1;
+struct WGPUNewStruct2;
+
+// Callback info structure forward declarations
+struct WGPUPrefixNewCallback1CallbackInfo;
+struct WGPUNewCallback2CallbackInfo;
+
+/**
+ * \defgroup Enumerations Enumerations
+ * \brief Enums.
+ *
+ * @{
+ */
+
+/**
+ * Extended enum shouldn't have the namespace prefix in the name by default.
+ */
+/**
+ * New enum entries that extend should have the prefix in the name by default.
+ */
+_wgpu_EXTEND_ENUM(WGPUOldEnum, WGPUOldEnum_PrefixNewValue, 0xFFFF0000);
+
+/**
+ * New enum should have the namespace prefix in the name by default.
+ */
+typedef enum WGPUPrefixNewEnum1 {
+    /**
+     * New enum entries for a new enum should not duplicate prefix.
+     */
+    WGPUPrefixNewEnum1_NewValue = 0xFFFF0000,
```

Comment:
This is larger than Force32, but we shouldn't allow that. In https://github.com/webgpu-native/webgpu-headers/blob/main/doc/articles/Extensions.md#registry-of-prefixes-and-enum-blocks it's noted enums are 32-bit and we reserve negative values.

(Enums are usually signed, so this might actually make the enum become a uint32_t or signed int64_t? It's implementation-defined.)

Reply:
Nice catch. Turned out to be a bug in the schema. The schema allowed the `enum_prefix` to be up to `2^16-1` which effectively allows setting the sign bit. Updated to only allow the maximum to be `2^15-1`.